init.lua

```diff
@@ -1,16 +1,35 @@
+local colors = {
+    green = {r = 0, g = 255, b = 0},
+    blue = {r = 0, g = 0, b = 255},
+    red = {r = 255, g = 0, b = 0},
+    purple = {r = 200, g = 0, b = 200},
+    yellow = {r = 255, g = 255, b = 0},
+    cyan = {r = 0, g = 255, b = 255},
+    white = {r = 255, g = 255, b = 255},
+    orange = {r = 255, g = 165, b = 0},
+    black = {r = 0, g = 0, b = 0},
+}
+
+-- Register privilege to allow players to colour their nametag
+minetest.register_privilege("nametag", {
+	description = "Allows you to colour your nametag",
+	give_to_singleplayer = false,
+})
+
+-- Register command to allow players to colour their nametag
 minetest.register_chatcommand("nametag",{
-	params = "<New Colour>",
-	description = "Alows a player to set a coloured nametag! This can either have a new colour as a param, or it can be used by itself to bring up a formspec.",
-	privs = {shout = true},
+	params = "<color>",
+	description = "Allows players to colour their nametag. Type /nametag to see available colors",
+	privs = {nametag = true},
```

Comment:
Was this requested by some players? Restricions should rather be done by running a callback, exposed as an API function (for example).

Reply:
> Was this requested by some players?

It was my idea, and I thought this would be a good idea.
TBH, I don't know if it could really be an improvement.

I can make it a setting or simply remove it.
What do you think?

Reply:
For now I'd say that `shout` suffices as restriction. If server owners want a better restriction, please either add a setting or let them override the chat command's `func` (to add further priv checks).